connectors/es/document.py

```diff
@@ -3,10 +3,86 @@
 # or more contributor license agreements. Licensed under the Elastic License 2.0;
 # you may not use this file except in compliance with the Elastic License 2.0.
 #
+from connectors.logger import logger
+
+
 class InvalidDocumentSourceError(Exception):
     pass
 
 
+class DocumentLogger:
+    def __init__(self, prefix, extra):
+        self._prefix = prefix
+        self._extra = extra
+
+    def isEnabledFor(self, level):
+        return logger.isEnabledFor(level)
+
+    def debug(self, msg, *args, **kwargs):
+        logger.debug(
+            msg,
+            *args,
+            prefix=self._prefix,  # pyright: ignore
```

Comment:
pyright is right, you can't pass `prefix` like this, it will error out. you want to create a new formatter

Reply:
pyright complains `No parameter named "prefix"`. Maybe what I should do is:
```python
def debug(self, msg, *args, **kwargs):
    kwargs["prefix"] = self._prefix
    kwargs["extra"] = self._extra
    logger.debug(
        msg,
        *args,
        **kwargs,
    )
```

But when I test the current code, it works

Reply:
```
>>> import logging
>>> logger = logging.getLogger('test')
>>> kw = {'prefix': 'ok'}
>>> logger.warn('message', **kw)
<stdin>:1: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/logging/__init__.py", line 1506, in warn
    self.warning(msg, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/logging/__init__.py", line 1501, in warning
    self._log(WARNING, msg, args, **kwargs)
TypeError: Logger._log() got an unexpected keyword argument 'prefix'
```

Reply:
Hi @tarekziade , I also made this change: https://github.com/elastic/connectors-python/pull/806/files#diff-a34ad6d1aa661c54359acc9e3e994d49e5b7b2d4f30f8db995682fd5d3f436c1

That's why it works.